packages/web/hooks/useImageUpload.ts

```diff
@@ -59,8 +60,22 @@ const useImageUpload = <T extends BaseUploadData>(getUploadData: () => Promise<T
     return result.value
   }
 
-  return [image, uploadingImage, onFileInputChange, () => setImage(null)]
-}
+  const onUnsplashSelect = (smallUrl: string, largeUrl: string) => {
```

Comment:
`useImageUpload` is shared between avatar, inline, and post image upload flows of which only post image upload involves unsplash integration so this probably isn't the place to add it. 

Also the type of `image` is generic here (because this hook is shared), which is why you need the `@ts-ignore` below.

I think the change you want to make here is to return `setImage` (as the last member of the return tuple so we don't have to refactor every consumer of this hook) and  then either in `usePostImageUplaod` or in the editor itself call `setImage` directly once the user has selected an item from unsplash.